### ISSUE TITLE
gen/golang: faster json marshalling

### ISSUE THREAD

**Issue by pkieltyka** [webrpc/webrpc#20]
As we have a well-defined schema, we have an opportunity to generate optimized JSON encoders/decoders for each different type that performs much better than Go's std "encoding/json". The std "encoding/json" uses the reflection to marshal/unmarshal in a flexible way, and its great, but produces extra garbage and costs more computer time then if we code-generated specific encoders for each message/type. See https://github.com/pquerna/ffjson as a viable tool we could use in gen/golang and offer 2x to 3x faster serialization performance by their benchmarks.

**Comment by VojtechVitek** [comment]
Very good blog post:
https://www.cockroachlabs.com/blog/high-performance-json-parsing/

**Comment by VojtechVitek** [comment]
Experimental feature with jsoniter:

```
webrpc-gen -schema=proto.ridl -target=golang -server -json=jsoniter -out=server.gen.go
webrpc-gen -schema=proto.ridl -target=golang -client -json=jsoniter -out=client.gen.go
```

It passed my initial testing against WebRPC interoperability test: https://github.com/webrpc/webrpc/tree/master/tests.

I'll add proper tests later.

**Comment by VojtechVitek** [comment]
Released in golang@v0.12.0
